waspc/examples/todoApp/ext/queries.js

```diff
@@ -34,7 +34,7 @@ export const getTask = async ({ id }, context) => {
     throw new HttpError(404)
   }
   if (task.user.id !== context.user.id) {
-    throw new HttpError(403)
+    throw new HttpError(401)
```

Comment:
Would we want this to remain a `403` @h4r1337? What this appears to be saying is we already have an authenticated user, but we are trying to get some task that belongs to a different user. So that feels a bit more Forbidden instead of Unauthorized to me. Wanted to double-check with you.

Reply:
That is a tricky use case. According to the [RFC 7235](https://datatracker.ietf.org/doc/html/rfc7235#section-3.1) :
> The 401 (Unauthorized) status code indicates that the request has not been applied because it lacks valid authentication credentials for the target resource...The user agent MAY repeat the request with a new or replaced Authorization header field.
In our case the target resource is the task belongs to a different user.

This is also explained in this [blog from auth0](https://auth0.com/blog/forbidden-unauthorized-http-status-codes/#Don-t-let-the-client-know---). So I think it is better to change it to `404`.

Reply:
Ah, interesting, `404` is pretty nice in that it obscures the existence due to lack of permissions and provides a bit more security than either `401`/`403` do. That is a good idea. Mind adding a comment in code to explain that we are trying to "to 'hide' the current existence of a forbidden target resource" in that case before we return `404`? Thanks.

Reply:
yea sure...